src/utility/utility.cpp

```diff
@@ -115,4 +116,60 @@ double EllipIntSecond(const double &x)
   return result;
 }
 
+bool almost_the_same(const double &a,
+                     const double &b,
+                     const double &rel_precision,
+                     const double &num_zero)
+{
+  if (std::abs(a) < num_zero and std::abs(b) < num_zero)
+  {
+    return true;
+  }
+  return std::abs(a - b) < std::abs(a + b) / 2 * rel_precision;
+}
+
+bool almost_the_same(const std::complex<double> &a,
+                     const std::complex<double> &b,
+                     const double &rel_precision,
+                     const double &num_zero)
+{
+  bool real_part = almost_the_same(a.real(), b.real(), rel_precision, num_zero);
+  bool imag_part = almost_the_same(a.imag(), b.imag(), rel_precision, num_zero);
+  return (real_part and imag_part);
+}
+
+bool almost_the_same(const std::vector<double> &a,
+                     const std::vector<double> &b,
+                     const bool &allow_for_sign_flip,
+                     const double &rel_precision,
+                     const double &num_zero)
+{
+  if (a.size() != b.size())
+  {
+    throw std::runtime_error("Error. Vectors must have the same size.");
```

Comment:
This case is not tested

Reply:
This case should now be tested for in `Test-gw.cpp`.